### PR TITLE
handle py2/py3 print statement vs function better (via user configurable option)

### LINKED ISSUE #119: Refactoring Python 2.7 files does not work (parse error on print statement)
https://pybowler.io/docs/basics-intro

> lib2to3 provides a concrete syntax tree (CST) implementation that recognizes and supports the grammar of all Python versions back to 2.6. 

> By building on lib2to3, Bowler is capable of reading and modifying source files written for both Python 2 and 3. That said, Bowler requires Python 3.6 or newer to run

Everything was working fine until I tried running on a (Py 2.7) source file which had a print statement in it

Then I get:

```
ERROR:RefactoringTool:Can't parse tests/fixtures/py27.py: ParseError: bad input: type=3, value='"Python 2 syntax"', context=(' ', (34, 10))
```

After putting some breakpoints in and digging through the source it seems the problem is here:
https://github.com/facebookincubator/Bowler/blob/195fbae4550ea82d72b9c59f3dca372b20d1edf5/bowler/tool.py#L98

`BowlerTool` hard-codes it so that the underlying fissix `RefactorTool` will use the no-print-statements grammar, which is therefore not Python 2 compatible.

Commenting out that line fixes it for me... I wonder why it was added?

### PR DESCRIPTION
**Pull request by anentropic**
fixes #119 

this allows Bowler to refactor py2 files containing `print` statements

### CODE REVIEW

File: bowler/tool.py [hunk 1]
```diff
@@ -148,6 +147,9 @@ def processed_file(
             if hunk:
                 hunks.append([a, b, *hunk])
 
+            features = _detect_future_features(new_text)
+            if "print_function" in features:
+                self.driver.grammar = pygram.python_grammar_no_print_statement
```
Comment (thatch) [inline bowler/tool.py]: I'm pretty sure this change will affect the following files as well.  If you look at the implementation of `refactor_string` in lib2to3 (which this is modeled after), it does a try/finally to restore it.

File: docs/api-query.md [hunk 2]
```diff
@@ -45,14 +45,23 @@ clarity and brevity.
...
+  filename_matcher: FilenameMatcher,
+)
 ```
 
 * `*paths` - Accepts either individual file or directory paths (relative to the current
   working directory), or lists of paths, for any positional argument given.
   Defaults to the current working directory if no arguments given.
 
-* `*filename_matcher*` - A callback which returns whether a given filename is
+* `python_version` - The 'major' python version of the files to be refactored, i.e. `2`
```
Comment (thatch) [inline docs/api-query.md]: Can you add asterisks back on the param names please?

Comment (anentropic) [reply docs/api-query.md]: sorry, I thought maybe they were intended as bold formatting but they display as literal asterisks (and it's not an *args param)

will put them back

also I will reorder docs and code to put my new arg after the existing one, that would be safer

File: bowler/tool.py (hunk 1 above)
Comment (anentropic) [reply bowler/tool.py]: ah right, I'll have a look and add this

Comment (anentropic) [reply bowler/tool.py]: done in ff8c0d4273be7f35141d04e407509717f53c34e1

File: docs/api-query.md (hunk 2 above)
Comment (anentropic) [reply docs/api-query.md]: done in ff8c0d4273be7f35141d04e407509717f53c34e1

Comment (thatch) [reply docs/api-query.md]: @jreese ?

File: bowler/tool.py (hunk 1 above)
Comment (thatch) [reply bowler/tool.py]: Thanks.

**Comment by jreese** [comment]
Thank you!

**Comment by anentropic** [comment]
Thanks so much for merging!

Do you have an idea when the next PyPI release will be?

**Comment by thatch** [comment]
I hope soon, but we ought to do some manual testing on windows first -- this release ought to work there (in single process mode) and I'd like to mention that in the release notes if so.

**Pull request merged**